connector/connect/client/jvm/pom.xml

```diff
@@ -75,6 +76,13 @@
       <artifactId>mockito-core</artifactId>
       <scope>test</scope>
     </dependency>
+    <!-- Use mima to perform the compatibility check -->
```

Comment:
Can we use SBT to check this instead of Maven? We have one place for MiMa so far in SBT (See also  `project/MimaBuild.scala`, and `dev/mima`)

Reply:
The SBT MiMa check has some limitations to run as a SBT rule:
It is the best for a stable API. e.g. current vs previous. It is not very friendly to configure to test e.g. scala-client vs sql while we are actively working on the scala-client API.
To be more specific, the problems I hit were:
1. I cannot configure the MiMa rule to find the current SQL SNAPSHOT jar.
2. I cannot use ClassLoader correctly in the SBT rule to load all methods in the client API.


As a result, I end up this test where we have more freedom to grow the API test coverage with the client API.

Reply:
Gotya. Let's probably add a couple of comments here and there to make it clear .. I am sure this is confusing to other developers.

Reply:
cc @dongjoon-hyun , also cc @pan3793 Do you have any suggestions for this?

Reply:
You can check out the MiMa SBT impl I did here: https://github.com/zhenlineo/spark/pull/6
I also marked the two problems in the PR code. Unless we can fix the two problems. I do not feel it is a better solution than this PR: calling MiMa directly in a test.